sql/core/src/main/scala/org/apache/spark/sql/RelationalGroupedDataset.scala

```diff
@@ -389,6 +389,16 @@ class RelationalGroupedDataset protected[sql](
   }
 
   /**
+   * Returns all grouping column names as an array.
+   *
+   * @since 2.1.0
+   */
+  def columns: Array[String] = {
+    val groupingNamedExpressions = groupingExprs.map(alias)
```

Comment:
This might not be a column (a named expression). Thus, this might not be useful to end users.